Sources/NostrSDK/Events/Channels/CreateChannelEvent.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  CreateChannelEvent.swift
+//
+//
+//  Created by Konstantin Yurchenko, Jr on 9/11/24.
+//
+
+import Foundation
+
+/// Create a public chat channel.
+/// See [NIP-28](https://github.com/nostr-protocol/nips/blob/master/28.md#kind-40-create-channel).
+public class CreateChannelEvent: NostrEvent {
```

Comment:
It would be good to expose the basic channel metadata (name, about, picture and relays) properties.

Take a look at https://github.com/nostr-sdk/nostr-sdk-ios/blob/main/Sources/NostrSDK/Events/MetadataEvent.swift as it also does JSON encoding and decoding for the kind 0 event.

Reply:
Should I make a MetadataChannel class here or what?

Reply:
In a separate file, you could create a `ChannelMetadata` struct with the properties. Then, you could create a `ChannelMetadataInterpreting` protocol that decodes the JSON.

Kinds 40 `CreateChannelEvent` and kind 41 `SetChannelMetadataEvent` could extend from `ChannelMetadataInterpreting`.

Something like this, it'll be pretty similar to what we do with kind 0:
```swift
public struct ChannelMetadata: Codable {
    public let name: String?
    public let about: String?
    ...

    enum CodingKeys: String, CodingKey {
        ...
    }
}

public protocol ChannelMetadataInterpreting: NostrEvent {}
public extension ChannelMetadataInterpreting {
    public var channelMetadata: ChannelMetadata {
        guard let data = content.data(using: .utf8) else {
            return nil
        }
        return try? JSONDecoder().decode(ChannelMetadata.self, from: data)
    }
}
```